thetis/callback.py

```diff
@@ -508,7 +508,7 @@ def __init__(self, solver_obj,
         """
         # printing all detector output to log is probably not a useful default:
         kwargs.setdefault('append_to_log', False)
-        self.field_dims = [solver_obj.fields[field_name].function_space().value_size
+        self.field_dims = [solver_obj.fields[field_name].function_space().block_size
```

Comment:
I'm a little confused tbh. I would have expected this should be `value_size` as that would correspond to the size of what you get when interpolating in a point (i.e. the size of the value) - which is not necessarily the same as the n/o DOFs per node which is what `block_size` gives IIUC. I know that your change should give back the old behaviour. Might it be that `DetectorsCallback` just has not been tested with RT or BDFM velocity fields?